internal/goroutine/periodic.go

```diff
@@ -40,6 +41,32 @@ type Finalizer interface {
 	OnShutdown()
 }
 
+// HandlerFunc wraps a function so it can be used as a Handler.
+type HandlerFunc func(ctx context.Context) error
+
+func (f HandlerFunc) Handle(ctx context.Context) error {
+	return f(ctx)
+}
+
+type simpleHandler struct {
+	name    string
+	handler func(ctx context.Context) error
```

Comment:
Maybe use the HandlerFunc type?

```suggestion
	handler HandlerFunc
```